.github/actions/report-ci-errors/action.yaml

```
@@ -0,0 +1,52 @@
+name: "Report CI errors to slack channel"
+description: "report CI errors to a nominated slack channel"
+inputs:
+  slack_webhook_url:
+    description: "Slack webhook URL"
+    required: true
+  channel:
+    description: "Name of the channel to send error notifications to"
+    required: true
+  github:
+    description: "Github Action context as json"
+    required: true
+  message:
+    description: "Optional message to show in channel"
+    default: ""
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        # run checks as github actions required is not being enforced - https://github.com/actions/runner/issues/1070
+        [[ "${{ inputs.slack_webhook_url }}" ]] || { echo "slack_webhook_url input is empty" ; exit 1; }
+        [[ "${{ inputs.channel }}" ]] || { echo "channel input is empty" ; exit 1; }
+      shell: bash
+    - name: Send slack
+      uses: slackapi/slack-github-action@v2.0.0
+      with:
+        errors: true
+        webhook: ${{ inputs.slack_webhook_url }}
+        webhook-type: incoming-webhook
+        payload: |
+          {
+            "channel": "${{ inputs.channel }}",
+            "text": "The <https://github.com/${{ fromJson(inputs.github).repository }}> CI test failed.",
```

The original request was for CI failures but it can be used to report any GitHub action run failure so I think we could make this message more generic.
```suggestion
            "text": "The <https://github.com/${{ fromJson(inputs.github).repository }}>'s GitHub workflow run failed.",
```

It would also be good if it included workflow name as repos tend to have a few. The acceptance criteria also mentions `branch` name but we decided it should be scoped to failures on `main` branch other wise it will be too noisy.

---

ok sure, that makes sense